src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,340 @@
+import com.google.api.codegen.util.java.JavaTypeTable;
+import com.google.api.codegen.viewmodel.ApiMethodType;
+import com.google.api.codegen.viewmodel.ApiMethodView;
+import com.google.api.codegen.viewmodel.PackageInfoView;
+import com.google.api.codegen.viewmodel.RetryCodesDefinitionView;
+import com.google.api.codegen.viewmodel.RetryParamsDefinitionView;
+import com.google.api.codegen.viewmodel.ServiceDocView;
+import com.google.api.codegen.viewmodel.SettingsDocView;
+import com.google.api.codegen.viewmodel.StaticApiMethodView;
+import com.google.api.codegen.viewmodel.StaticXApiView;
+import com.google.api.codegen.viewmodel.StaticXSettingsView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.gax.core.RetrySettings;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import io.grpc.Status.Code;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map.Entry;
+
+/**
+ * The ModelToViewTransformer to transform a Model into the standard GAPIC surface in Java.
+ */
+public class JavaGapicSurfaceTransformer implements ModelToViewTransformer {
+  private GapicCodePathMapper pathMapper;
+  private ServiceTransformer serviceTransformer;
+  private PathTemplateTransformer pathTemplateTransformer;
+  private ApiCallableTransformer apiCallableTransformer;
+  private ApiMethodTransformer apiMethodTransformer;
+  private PageStreamingTransformer pageStreamingTransformer;
+  private BundlingTransformer bundlingTransformer;
+
+  private static final String XAPI_TEMPLATE_FILENAME = "java/main.snip";
+  private static final String XSETTINGS_TEMPLATE_FILENAME = "java/settings.snip";
+  private static final String PACKAGE_INFO_TEMPLATE_FILENAME = "java/package-info.snip";
+
+  /**
+   * Standard constructor.
+   */
+  public JavaGapicSurfaceTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+    this.serviceTransformer = new ServiceTransformer();
+    this.pathTemplateTransformer = new PathTemplateTransformer();
+    this.apiCallableTransformer = new ApiCallableTransformer();
+    this.apiMethodTransformer = new ApiMethodTransformer();
+    this.pageStreamingTransformer = new PageStreamingTransformer();
+    this.bundlingTransformer = new BundlingTransformer();
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(
+        XAPI_TEMPLATE_FILENAME, XSETTINGS_TEMPLATE_FILENAME, PACKAGE_INFO_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new JavaSurfaceNamer();
```

Comment:
True, but I think it's more flexible for future maintenance to go through SurfaceNamer. It allows multiple languages to use the same patterns if it makes sense to do so in the future. It also means there is one reference point for all of the naming patterns that we have (we just have a small set of language-specific overrides).